Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -164,6 +176,52 @@ public struct NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Sendable {
             outboundWriter: outboundWriter
         )
     }
+
+    /// Provides scoped access to the inbound and outbound side of the underlying ``Channel``.
+    ///
+    /// - Important: After this method returned the underlying ``Channel`` will be closed.
+    ///
+    /// - Parameter body: A closure that gets scoped access to the inbound and outbound.
+    public func withInboundOutbound<Result>(
+        _ body: (NIOAsyncChannelInboundStream<Inbound>, NIOAsyncChannelOutboundWriter<Outbound>) async throws -> Result
```

Comment:
I didn't ask for external label ;)

The instructions in the API guidelines say should have internal labels because they act as documentation and act as hints in auto completion etc.

Reply:
Oh sorry, I misunderstood you then. Pushed a commit with internal labels.